activerecord/lib/active_record/database_configurations/connection_url_resolver.rb

```diff
@@ -38,7 +38,11 @@ def initialize(url)
       # Converts the given URL to a full connection hash.
       def to_hash
         config = raw_config.compact_blank
-        config.map { |key, value| config[key] = uri_parser.unescape(value) if value.is_a? String }
+        config.map do |key, value|
+          next unless value.is_a? String
+
+          config[key] = value.downcase == "false" ? false : uri_parser.unescape(value)
```

Comment:
It feels awkward to me that we only handle false here. Can you take a look at how we do this elsewhere in Rails and see if there's a solution that handles both true and false boolean values?

Reply:
Thanks for the comments!

I don't believe there is a similar case elsewhere in the codebase. The reason for handling only false values is that any other string will evaluate as true anyway - yes, it could be explicitly handled, but it would just add another line of code.

Another alternative would be to skip `compact_blank` in #L40, allowing nil as false, no special handling at all. It does not allow the IMO intuitive use of passing `param=false` however. 

Numeric values are converted eg in [hash_config.rb](https://github.com/rails/rails/blob/a1367da11dbe1fb9984c7e90b1bb9dab96977948/activerecord/lib/active_record/database_configurations/hash_config.rb#L66) but you cannot do the same with boolean

Reply:
Eg a url_config approach perhaps like:
```ruby
private
  BOOOLEAN_KEYS = %i[replica schema_dump database_tasks].freeze
                                                                          
  private_constant :BOOOLEAN_KEYS
                                                                          
  # Return a Hash that can be merged into the main config that represents
  # the passed in url
  def build_url_hash
    if url.nil? || url.start_with?("jdbc:", "http:", "https:")
      { url: url }
    else
      convert_boolean_keys(ConnectionUrlResolver.new(url).to_hash)
    end
  end
                                                                          
  def convert_boolean_keys(hash)
    Hash[hash.map { |k, v| [k, k.in?(BOOOLEAN_KEYS) ? bool(v) : v] }]
  end
                                                                          
  def bool(value)
    case value.downcase
    when "false", "f"
      false
    else
      value
    end
  end

```

Reply:
Hi @eileencodes! Did you get a chance to consider the alternative approaches suggested? If you think one of them would look nicer I can change the implementation